src/components/ImageUploader/index.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import Icon from '../Icon';
+import IconButton from '../IconButton';
+import UploaderImageItem from '../UploaderImageItem';
+
+import { toBase64 } from '../../utils';
+import { StyledImageUploader, UploaderPreview } from './styles';
+
+const getDefaultValuePreviews = files =>
+  Array.isArray(files) ? files.map(file => file.preview) : [];
+
+const ImageUploader = React.forwardRef(
+  ({ accept, className, multiple, onChange, value, ...props }, ref) => {
+    const [files, setFiles] = React.useState(value);
+    const [preview, setPreview] = React.useState(
+      getDefaultValuePreviews(value),
+    );
+
+    const [isLoading, setIsLoading] = React.useState(false);
+    const [hasError, setHasError] = React.useState(false);
+
+    const handleDeleteFile = fileToDelete => {
+      const filteredFiles = files.filter(
+        file => file.name !== fileToDelete.name,
+      );
+
+      setFiles(filteredFiles);
+      onChange(filteredFiles);
+    };
+
+    const handleFileChange = inputFiles => {
+      const nextFiles = multiple ? [...files, ...inputFiles] : inputFiles;
+
+      setFiles(nextFiles);
+      onChange(inputFiles);
+
+      if (multiple) {
+        return;
+      }
+
+      setIsLoading(true);
+
+      toBase64(inputFiles[0])
+        .then(base64 => {
+          setPreview(base64);
+          setIsLoading(false);
+        })
+        .catch(setHasError);
+    };
+
+    const handleClear = () => {
+      setPreview([]);
+      setFiles([]);
+      onChange([]);
+    };
+
+    const hasPreview = !multiple && preview.length > 0;
+
+    return (
+      <StyledImageUploader
+        className={cx('f-ImageUploader', className)}
+        onChange={handleFileChange}
+        multiple={multiple}
+        hasPreview={hasPreview}
+        isLoading={isLoading}
+        hasError={hasError}
+        overrides={{
+          input: { accept },
+        }}
+        ref={ref}
+        value={files}
+        {...props}
+      >
+        {hasPreview && (
+          <>
+            <UploaderPreview
+              className='f-ImageUploader-preview'
+              preview={preview}
+            />
+
+            <div className='f-ImageUploader-hoverState'>
```

Comment:
Shouldn't this be a component by itself?

Reply:
Do you know if you're using these classes anywhere? I'd like to remove them but I'm not sure if they're being used. We're not using `f-ImageUploader-hoverState` or `f-ImageUploader-preview` in the admin.

They're not that important, if I can remove them it's cleaner as a SC